File: algorithms/fibonacci/javascript/fibonacci.js
function fibonacci(n) {
  if (n===1) {
    return [0, 1];
  } else {
    var s = fibonacci(n - 1);
    s.push(s[s.length - 1] + s[s.length - 2]);
    return s;
  }
};

 console.log(fibonacci(8));
 

File: algorithms/fibonacci/java/Fibonacci.java
import java.util.stream.Stream;

/**
 * A Java implementation of Fibonacci
 */
public class Fibonacci {

    public static int fibonacci(final int index){
        int firstOnOld=0;
        int secondOnHold=1;

        if (index == 0){
            return firstOnOld;
        }else if (index == 1) {
            return secondOnHold;
        }

        for (int i=2;i<=index;i++){
            int current = firstOnOld + secondOnHold;
            firstOnOld=secondOnHold;
            secondOnHold=current;
        }

        return secondOnHold;
    }

    /**
     * Test
     * @param args - arg
     */
    public static void main (String[] args) {
        Stream.iterate(0, n -> n<20, n-> n+1).forEach(x-> System.out.println("fibo of " + x +" is " +fibonacci(x)));
    }
}
